pwndbg/gdblib/proc.py

```diff
@@ -103,6 +105,44 @@ def binary_base_addr(self) -> int:
     def binary_vmmap(self) -> Tuple[pwndbg.lib.memory.Page, ...]:
         return tuple(p for p in pwndbg.gdblib.vmmap.get() if p.objfile == self.exe)
 
+    @pwndbg.lib.memoize.reset_on_start
+    @pwndbg.lib.memoize.reset_on_objfile
+    def dump_elf_data_section(self) -> Optional[Tuple[int, int, bytes]]:
+        """
+        Dump .data section of current process's ELF file
+        """
+        local_path = pwndbg.gdblib.file.get_file(pwndbg.gdblib.proc.exe)
+        with open(local_path, "rb") as f:
+            elffile = ELFFile(f)
+            section = elffile.get_section_by_name(".data")
+            if section:
+                return section["sh_addr"], section["sh_size"], section.data()
```

Comment:
perhaps we can move this to some gdblib.file or gdblib.elf function; I mean we could create a func that returns given section addr, size and data

Reply:
Ok, I'll move some of the code into `gdblib.elf`.